pkg/virt-controller/watch/clone/util.go

```diff
@@ -34,42 +35,15 @@ func getKey(name, namespace string) string {
 	return fmt.Sprintf("%s/%s", namespace, name)
 }
 
-func generateNameWithRandomSuffix(names ...string) string {
-	const randomStringLength = 5
-
-	if len(names) == 0 {
-		return ""
-	}
-
-	generatedName := names[0]
-	for _, name := range names[1:] {
-		generatedName = fmt.Sprintf("%s-%s", generatedName, name)
-	}
-
-	// Kubernetes' object names have limit of 252 characters.
-	// For more info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
-	if len(generatedName) > 252 {
-		generatedName = "clone-object"
-	}
-
-	generatedName = fmt.Sprintf("%s-%s", generatedName, rand.String(randomStringLength))
```

Comment:
I'm hesitant about removing the random generated suffixes for these names; generally if a name isn't provided when creating something in KubeVirt, it will generate a name with a randomly generated suffix in this manner (also true when specifying a `GenerateName` in its ObjectMeta.

Reply:
You're right, while the issue will appear if we use random suffix. I find resource in export package
```
//pkg/storage/export/export/export.go
// getDefaultTokenSecretName returns a secret name specifically created for the current export object
func getDefaultTokenSecretName(vme *exportv1.VirtualMachineExport) string {
	return naming.GetName("export-token", vme.Name, validation.DNS1035LabelMaxLength)
}

func (ctrl *VMExportController) getExportServiceName(vmExport *exportv1.VirtualMachineExport) string {
	return naming.GetName(exportPrefix, vmExport.Name, validation.DNS1035LabelMaxLength)
}

func (ctrl *VMExportController) getExportPodName(vmExport *exportv1.VirtualMachineExport) string {
	return naming.GetName(exportPrefix, vmExport.Name, validation.DNS1035LabelMaxLength)
}
```
all use GetName method. It create suffix with hash. 
In my opinion, use this method to generate name is ok, and we can use it to temporarily fix this problem which is likely to occur.